include/rx/ranges.hpp

```diff
@@ -1655,6 +1656,107 @@ struct reverse {
     }
 };
 
+/// Sliding window over an input range.
+///
+/// E.g. for a range `(1,2,3,4,5)` a window of 2 elements is `[(1,2), (2,3), (3,4), (5)]`.
+/// For a range `(1,2,3,4,5,6)` a window of 2 elements with a stepsize of 3 is `[(1,2), (4,5)]`.
+///
+/// Inspired by [more_itertools.windowed()](https://more-itertools.readthedocs.io/en/v7.2.0/api.html#more_itertools.windowed).
+struct windowed {
+    size_t window_size_ = 1;
+    size_t step_size_ = 1;
+
+    explicit constexpr windowed(size_t window_size, size_t step_size = 1) noexcept
+        : window_size_(window_size), step_size_(step_size) {
+        RX_ASSERT(window_size > 0);
+        RX_ASSERT(step_size > 0);
+    }
+    constexpr windowed(const windowed&);
```

Comment:
You are missing the definition of it. Though, I'm confused why you even declare them as the implicitly declared copy/move constructors/assignment operators are already constexpr and noexcept.

Reply:
Thank you for reviewing my code! If I have a user defined constructor, the implicit constructors aren't automatically generated, are they? I'll add the missing `= default`.

Reply:
They are still implicitly defined: https://en.cppreference.com/w/cpp/language/copy_constructor#Deleted_implicitly-declared_copy_constructor

Reply:
Thank you for the link! Lazy as I am I never really bothered to remember when I *have to* add the c'tors. I'll simply always add all five lines:

```c++
constexpr T() noexcept = default;
constexpr T(const T&) noexcept = default;
constexpr T(T&&) noexcept = default;
constexpr T &operator =(const T&) noexcept = default;
constexpr T &operator =(T&&) noexcept = default;
```